File: codes/leetcode/HouseRobber.cpp
/*
************************************************
username    :   smmehrab
fullname    :   s.m.mehrabul islam
email       :   mehrab.24csedu.001@gmail.com
institute   :   university of dhaka, bangladesh
session     :   2017-2018
************************************************
*/
class Solution {
public:
    int rob(vector<int>& houses) {
        int currentMax = 0, previousMax = 0, newMax;
        for(int money : houses){
            newMax = max(currentMax, previousMax + money);
            previousMax = currentMax;
            currentMax = newMax;
        }

        return currentMax;
    }
};

File: codes/leetcode/HouseRobber.py
# ************************************************
# username  :   smmehrab
# fullname  :   s.m.mehrabul islam
# email     :   smmehrabul-2017614964@cs.du.ac.bd
# institute :   university of dhaka, bangladesh
# reg       :   2017614964
# ************************************************

class Solution:
    def rob(self, nums: list[int]) -> int:
        n = len(nums)
        if n>1:
            nums[1] = max(nums[0], nums[1])
            for i in range(2, n):
                nums[i] = max(nums[i]+nums[i-2], nums[i-1])
        return nums[n-1]
